reactor-netty-core/src/main/java/reactor/netty/transport/ProxyProvider.java

```diff
@@ -265,28 +269,39 @@ static ProxyProvider createFrom(Properties properties) {
 	static ProxyProvider createHttpProxyFrom(Properties properties) {
 		String hostProperty;
 		String portProperty;
+		String userProperty;
+		String passwordProperty;
 		String defaultPort;
 		if (properties.containsKey(HTTPS_PROXY_HOST)) {
 			hostProperty = HTTPS_PROXY_HOST;
 			portProperty = HTTPS_PROXY_PORT;
+			userProperty = HTTPS_PROXY_USER;
+			passwordProperty = HTTPS_PROXY_PASSWORD;
 			defaultPort = "443";
 		}
 		else {
 			hostProperty = HTTP_PROXY_HOST;
 			portProperty = HTTP_PROXY_PORT;
+			userProperty = HTTP_PROXY_USER;
+			passwordProperty = HTTP_PROXY_PASSWORD;
 			defaultPort = "80";
 		}
 
 		String hostname = Objects.requireNonNull(properties.getProperty(hostProperty), hostProperty);
 		int port = parsePort(properties.getProperty(portProperty, defaultPort), portProperty);
 
+		String username = properties.getProperty(userProperty);
+		String password = properties.getProperty(passwordProperty);
+
 		String nonProxyHosts = properties.getProperty(HTTP_NON_PROXY_HOSTS, DEFAULT_NON_PROXY_HOSTS);
 		RegexShouldProxyPredicate transformedNonProxyHosts = RegexShouldProxyPredicate.fromWildcardedPattern(nonProxyHosts);
 
 		return ProxyProvider.builder()
 				.type(ProxyProvider.Proxy.HTTP)
 				.host(hostname)
 				.port(port)
+				.username(username)
```

Comment:
here I think we need to conditionally invoke `username` and `password` methods? (the same way it is done for the SOCKS proxy configuration below)

Reply:
I thought about doing that, but it appears that the username and password methods are just "setters", so passing null is just a no-op and doesn't break anything. If you think this behavior is subject to change, it might be worthwile to add a nullity check. Please let me know what you think is best.